### PR TITLE
Standardize spelling of speech-to-text throughout docs

### PR DESCRIPTION
**Pull request by c0ffeeca7**
## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_posts/2019-11-20-release-102.markdown [hunk 1]
```diff
@@ -112,7 +112,7 @@ Users of Xiaomi Vacuum cleaners can now use this great custom Lovelace card:
 
 - Add Unifi Led ([@florisvdk] - [#27475]) ([unifiled docs]) (new-integration)
 - Almond integration ([@balloob] - [#28282]) ([almond docs]) (new-integration)
-- Speech to Text component ([@pvizeli] - [#28434]) ([demo docs]) ([stt docs]) (new-integration)
+- Speech-to-text component ([@pvizeli] - [#28434]) ([demo docs]) ([stt docs]) (new-integration)
```
Comment (frenck) [inline source/_posts/2019-11-20-release-102.markdown]: These are commit messages, we should not change those.

File: source/_posts/2019-11-20-release-102.markdown [hunk 2]
```diff
@@ -381,7 +381,7 @@ Experiencing issues introduced by this release? Please report them in our [issue
 - Remove legacy reproduce state ([@balloob] - [#28458])
 - Align attribute naming between light and switch for HomematicIP Cloud ([@SukramJ] - [#28271]) ([homematicip_cloud docs]) (breaking change)
 - Enable transition time for HmIP-BSL - HomematicIP Cloud ([@SukramJ] - [#28201]) ([homematicip_cloud docs])
-- Speech to Text component ([@pvizeli] - [#28434]) ([demo docs]) ([stt docs]) (new-integration)
+- Speech-to-text component ([@pvizeli] - [#28434]) ([demo docs]) ([stt docs]) (new-integration)
```
Comment (frenck) [inline source/_posts/2019-11-20-release-102.markdown]: same

File: source/changelogs/core-2023.4.markdown [hunk 3]
```diff
@@ -578,7 +578,7 @@ For a summary in a more readable format:
 - Prepare MQTT platform tests part4 ([@jbouwh] - [#90107])
 - Prepare MQTT platform tests part8 ([@jbouwh] - [#90132])
 - Prepare MQTT platform tests part9 ([@jbouwh] - [#90133])
-- Add speech to text over binary websocket to pipeline ([@synesthesiam] - [#90082])
+- Add speech-to-text over binary websocket to pipeline ([@synesthesiam] - [#90082])
```
Comment (frenck) [inline source/changelogs/core-2023.4.markdown]: same

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_posts/2019-11-20-release-102.markdown (hunk 1 above)
Comment (c0ffeeca7) [reply source/_posts/2019-11-20-release-102.markdown]: Ah, I see! Thank you for the catch!

Comment (c0ffeeca7) [inline source/_posts/2019-11-20-release-102.markdown]:
```suggestion
- Speech to Text component ([@pvizeli] - [#28434]) ([demo docs]) ([stt docs]) (new-integration)
```

File: source/_posts/2019-11-20-release-102.markdown (hunk 2 above)
Comment (c0ffeeca7) [inline source/_posts/2019-11-20-release-102.markdown]:
```suggestion
- Speech to Text component ([@pvizeli] - [#28434]) ([demo docs]) ([stt docs]) (new-integration)
```

File: source/changelogs/core-2023.4.markdown (hunk 3 above)
Comment (c0ffeeca7) [inline source/changelogs/core-2023.4.markdown]:
```suggestion
- Add speech to text over binary websocket to pipeline ([@synesthesiam] - [#90082])
```

**Review by c0ffeeca7 — COMMENTED** [review]
Undo changes on commit messages

File: source/_integrations/google_cloud.markdown [hunk 4]
```diff
@@ -42,10 +42,11 @@ Basic instruction for all APIs:
 4. [Make sure that billing is enabled for your Google Cloud Platform project](https://cloud.google.com/billing/docs/how-to/modify-project).
 5. Enable needed Cloud API visiting one of the links below or [APIs library](https://console.cloud.google.com/apis/library), selecting your `Project` from the dropdown list and clicking the `Continue` button:
 
+
     * [Text-to-speech](https://console.cloud.google.com/flows/enableapi?apiid=texttospeech.googleapis.com)
-    * [Speech-to-Text](https://console.cloud.google.com/flows/enableapi?apiid=speech.googleapis.com)
+    * [Speech-to-text](https://console.cloud.google.com/flows/enableapi?apiid=speech.googleapis.com)
     * [Geocoding](https://console.cloud.google.com/flows/enableapi?apiid=geocoding-backend.googleapis.com)
-
+ssssssssssssssssssssssssssssssssssss
```
Comment (frenck) [inline source/_integrations/google_cloud.markdown]: Do you have a cat? 😸

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/google_cloud.markdown (hunk 4 above)
Comment (c0ffeeca7) [reply source/_integrations/google_cloud.markdown]: Finally. There is evidence of the coffeecat.

Comment (c0ffeeca7) [inline source/_integrations/google_cloud.markdown]:
```suggestion
```

**Review by c0ffeeca7 — COMMENTED** [review]
cat on keyboard. dust off foot prints

**Review by frenck — APPROVED** [review]
Thanks, @c0ffeeca7 👍 

../Frenck

File: source/_posts/2022-12-20-year-of-voice.markdown [hunk 5]
```diff
@@ -44,7 +44,7 @@ With Home Assistant we want to make a privacy and locally focused smart home ava
 
 With Home Assistant we prefer to get the things we’re building in the user's hands as early as possible. Even basic functionality allows users to find things that work and don’t work, allowing us to address the direction if needed.
 
-A voice assistant has a lot of different parts: hot word detection, speech to text, intent recognition, intent execution, text-to-speech. Making each work in every language is a lot of work. The most important part is the intent recognition and intent execution. We need to be able to understand your commands and execute them.
+A voice assistant has a lot of different parts: hot word detection, speech-to-text, intent recognition, intent execution, text to speech. Making each work in every language is a lot of work. The most important part is the intent recognition and intent execution. We need to be able to understand your commands and execute them.
```
Comment (MartinHjelmare) [inline source/_posts/2022-12-20-year-of-voice.markdown:47]: I don't think we should have touched `text-to-speech`. A rebase went wrong?

Comment (c0ffeeca7) [reply source/_posts/2022-12-20-year-of-voice.markdown:47]: quite possibly... Thank you for the catch! #27613

Comment (c0ffeeca7) [reply source/_posts/2022-12-20-year-of-voice.markdown:47]: quite possibly... Thank you for the catch! #27613

**Pull request merged**